lib/metadata/schema-parser.js

```diff
@@ -698,29 +765,136 @@ SchemaParserV2.prototype._parseTableOrView = function (tableInfo, encoder, table
-    tableInfo.includeAllColumns = tableRow['include_all_columns'];
-    return tableInfo;
-  }
-  tableInfo.indexes = Index.fromRows(indexRows);
-  var flags = tableRow['flags'];
-  var isDense = flags.indexOf('dense') >= 0;
-  var isSuper = flags.indexOf('super') >= 0;
-  var isCompound = flags.indexOf('compound') >= 0;
-  tableInfo.isCompact = isSuper || isDense || !isCompound;
-  //remove the columns related to Thrift
-  var isStaticCompact = !isSuper && !isDense && !isCompound;
-  if(isStaticCompact) {
-    pruneStaticCompactTableColumns(tableInfo);
-  }
-  else if (isDense) {
-    pruneDenseTableColumns(tableInfo)
-  }
-  return tableInfo;
+    if (isView) {
+      tableInfo.tableName = tableRow['base_table_name'];
+      tableInfo.whereClause = tableRow['where_clause'];
+      tableInfo.includeAllColumns = tableRow['include_all_columns'];
+      return callback();
+    }
+    tableInfo.indexes = Index.fromRows(indexRows);
+    var flags = tableRow['flags'];
+    var isDense = flags.indexOf('dense') >= 0;
+    var isSuper = flags.indexOf('super') >= 0;
+    var isCompound = flags.indexOf('compound') >= 0;
+    tableInfo.isCompact = isSuper || isDense || !isCompound;
+    //remove the columns related to Thrift
+    var isStaticCompact = !isSuper && !isDense && !isCompound;
+    if(isStaticCompact) {
+      pruneStaticCompactTableColumns(tableInfo);
+    }
+    else if (isDense) {
+      pruneDenseTableColumns(tableInfo)
+    }
+    callback();
+  });
+};
+
+/** @override */
+SchemaParserV2.prototype._parseAggregate = function (row, callback) {
+  var encoder = this.cc.getEncoder();
+  var aggregate = new Aggregate();
+  aggregate.name = row['aggregate_name'];
+  aggregate.keyspaceName = row['keyspace_name'];
+  aggregate.signature = row['argument_types'] || utils.emptyArray;
+  aggregate.stateFunction = row['state_func'];
+  aggregate.finalFunction = row['final_func'];
+  aggregate.initConditionRaw = row['initcond'];
+  var self = this;
+  async.series([
+    function parseArguments(next) {
+      async.map(row['argument_types'] || utils.emptyArray, function (name, mapNext) {
+        encoder.parseTypeName(row['keyspace_name'], name, 0, null, self.udtResolver, mapNext);
+      }, function (err, result) {
+        aggregate.argumentTypes = result;
+        next(err);
+      });
+    },
+    function parseStateType(next) {
+      encoder.parseTypeName(row['keyspace_name'], row['state_type'], 0, null, self.udtResolver, function (err, type) {
+        aggregate.stateType = type;
+        next(err);
+      });
+    },
+    function parseReturnType(next) {
+      encoder.parseTypeName(row['keyspace_name'], row['return_type'], 0, null, self.udtResolver, function (err, type) {
+        aggregate.returnType = type;
+        next(err);
+      });
+    }
+  ], function (err) {
+    if (err) {
+      return callback(err);
+    }
+    try {
+      aggregate.initCondition = encoder.decode(aggregate.initConditionRaw, aggregate.stateType);
```

Comment:
It looks like it will not work as I suspected since `aggregate.initConditionRaw` is a string, get a TypeError at this line:

```
atolbert@atolbert-rmbp:~/Documents/Projects/nodejs-driver$ mocha test/integration/short/udf-tests.js -g 'should retrieve the most up to date metadata'


  Metadata
	...Starting test C* cluster v3.0.0 with 1 node(s)
	...With --install-dir=/Users/atolbert/Documents/Projects/cassandra
    #getFunction()
      ✓ should retrieve the most up to date metadata (5197ms)
    #getAggregate()
      1) should retrieve the most up to date metadata


  1 passing (17s)
  1 failing

  1) Metadata #getAggregate() should retrieve the most up to date metadata:
     Uncaught TypeError: bytes.readInt32BE is not a function
      at Encoder.decodeInt (lib/encoder.js:443:16)
      at Encoder.decode (lib/encoder.js:170:18)
      at lib/metadata/schema-parser.js:829:41
      at node_modules/async/lib/async.js:254:17
      at node_modules/async/lib/async.js:157:25
      at node_modules/async/lib/async.js:251:21
      at node_modules/async/lib/async.js:615:34
      at lib/metadata/schema-parser.js:821:9
      at Encoder.parseTypeName (lib/encoder.js:1157:12)
      at parseReturnType (lib/metadata/schema-parser.js:819:15)
      at node_modules/async/lib/async.js:610:21
      at node_modules/async/lib/async.js:249:17
      at iterate (node_modules/async/lib/async.js:149:13)
      at node_modules/async/lib/async.js:160:25
      at node_modules/async/lib/async.js:251:21
      at node_modules/async/lib/async.js:615:34
      at lib/metadata/schema-parser.js:815:9
      at Encoder.parseTypeName (lib/encoder.js:1157:12)
      at parseStateType (lib/metadata/schema-parser.js:813:15)
      at node_modules/async/lib/async.js:610:21
      at node_modules/async/lib/async.js:249:17
      at iterate (node_modules/async/lib/async.js:149:13)
      at node_modules/async/lib/async.js:160:25
      at node_modules/async/lib/async.js:251:21
      at node_modules/async/lib/async.js:615:34
      at lib/metadata/schema-parser.js:809:9
      at node_modules/async/lib/async.js:254:17
      at done (node_modules/async/lib/async.js:135:19)
      at node_modules/async/lib/async.js:32:16
      at node_modules/async/lib/async.js:251:21
      at Encoder.parseTypeName (lib/encoder.js:1157:12)
      at async.map.aggregate.argumentTypes (lib/metadata/schema-parser.js:806:17)
```